test/Codec/Binary/Bech32Spec.hs

```diff
@@ -607,6 +695,50 @@ instance Arbitrary HumanReadablePart where
       where
         chars = humanReadablePartToText hrp
 
+-- | A human-readable part that may (or may not) contain one or more invalid
+--   characters.
+newtype HumanReadablePartWithSuspiciousChars =
+    HumanReadablePartWithSuspiciousChars Text
+    deriving (Eq, Show)
+
+instance Arbitrary HumanReadablePartWithSuspiciousChars where
+    arbitrary = do
+        len <- genLength
+        chars <- replicateM len genChar
+        return $ HumanReadablePartWithSuspiciousChars $ T.pack chars
+      where
+        genLength = choose
+            ( humanReadablePartMinLength
+            , humanReadablePartMaxLength
+            )
+        genChar = choose
+            ( pred humanReadableCharMinBound
```

Comment:
so here there are two characters invalid : `pred humanReadableCharMinBound` and `succ humanReadableCharMaxBound`. The rest is valid, ie., any character from <humanReadableCharMinBound, humanReadableCharMaxBound>   I am not saying this is wrong, because it is very ok. but maybe we can extend the set of invalid characters here? (just ntpicking)

Reply:
>  I am not saying this is wrong, because it is very ok. but maybe we can extend the set of invalid characters here? (just nitpicking)

Done. See: https://github.com/input-output-hk/bech32/pull/7/commits/85e4e937a7aac87b3aa026b3059e3d9299c85d46